File: Python_Solutions/selection_sort/selection_sort.py
# Implement a selection sort algorithm

def selection_sort(arr):
    for i in range(len(arr) - 1):
        lowest = i
        for j in range(i+1, len(arr), 1):
            if arr[j] < arr[lowest]:
                lowest = j
        if i != lowest:
            temp = arr[i]
            arr[i] = arr[lowest]
            arr[lowest] = temp
    return arr


print(selection_sort([4, 1, 2, 51, 24, 56, 32, 12, 4,
      7, 100, 9, -1, -12, 21, 30, 34, -3, -21, 31, 34]))

File: Js_Solutions/selection_sort/selection_sort.js
// Implement a selection sort algorithm

function selectionSort(arr) {
    for (let i = 0; i < arr.length - 1; i++) {
        let min = i;
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[min]) {
                min = j;
            }
        }
        if (i !== min) {
            let temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
        }
    }
    return arr;
}

console.log(selectionSort([4, 1, 2, 51, 24, 56, 32, 12, 4, 7, 100, 9, -1, -12, 21, 30, 34, -3, -21, 31, 34]));